core/silkworm/state/in_memory_state.cpp

```diff
@@ -44,19 +44,17 @@ ByteView InMemoryState::read_code(const evmc::bytes32& code_hash) const noexcept
 
 evmc::bytes32 InMemoryState::read_storage(const evmc::address& address, uint64_t incarnation,
                                           const evmc::bytes32& location) const noexcept {
-    auto it1{storage_.find(address)};
-    if (it1 == storage_.end()) {
-        return {};
-    }
-    auto it2{it1->second.find(incarnation)};
-    if (it2 == it1->second.end()) {
-        return {};
-    }
-    auto it3{it2->second.find(location)};
-    if (it3 == it2->second.end()) {
-        return {};
+    const auto it1{storage_.find(address)};
+    if (it1 != storage_.end()) {
+        const auto it2{it1->second.find(incarnation)};
+        if (it2 != it1->second.end()) {
+            const auto it3{it2->second.find(location)};
+            if (it3 != it2->second.end()) {
+                return it3->second;
+            }
+        }
```

Comment:
I personally like the version of the left better, but if we use this style why not have the declarations inside the if test?

```
    if (const auto it1{storage_.find(address)}; it1 != storage_.end()) {
        if (const auto it2{it1->second.find(incarnation)}; it2 != it1->second.end()) {
            if (const auto it3{it2->second.find(location)}; it3 != it2->second.end()) {
                return it3->second;
            }
        }
    }
```